src/lib.rs

```diff
@@ -377,24 +377,10 @@ where
         let mut extensions = span.extensions_mut();
 
         if extensions.get_mut::<Map<String, Value>>().is_none() {
-            let mut object: Map<String, Value> = self.base_object.clone();
-            let mut add_field_visitor = visitor::AdditionalFieldVisitor::new(&mut object);
-            attrs.record(&mut add_field_visitor);
-            extensions.insert(object)
-        }
-    }
-
-    fn on_record(&self, id: &Id, values: &Record<'_>, ctx: Context<S>) {
```

Comment:
I think on record still needs to exist because it is the handler if I want to add a field to a span after it has been created. Without this any fields added by looking up the current span and calling record won't be saved unless of course I'm mistaken.

Reply:
I'll add this back in, I think, much like the above, we don't need to clone the base object?

Reply:
We do need to get the data that is in the base object when we create a new span or if the extensions aren't there in the record scenario because you would want the base object fields to be recorded as well. I might be missing something. @hlb8122

Reply:
@glademiller I've re-added the `on_record`, but without the `self.base_object.clone()`.

```rust
use std::net::SocketAddr;
use tracing_gelf::Logger;

#[tokio::main]
async fn main() {
    // Graylog address
    let address: SocketAddr = "127.0.0.1:12201".parse().unwrap();

    // Start tracing
    let bg_task = Logger::builder()
        .additional_field("ADD", "FEILD")
        .init_tcp(address)
        .unwrap();

    // Spawn background task
    // Any futures executor can be used
    tokio::spawn(bg_task);

    // Create a span
    let span = tracing::info_span!("CAVE A", testa = "SPAN A", morea = "HEY A");
    span.in_scope(|| {
        // Create a span
        tracing::warn!(message = "INSIDE THE CAVE A");

        let span = tracing::info_span!("CAVE B", testb = "SPAN B", moreb = "HEY B");
        span.in_scope(|| {
            // Log inside a span
            tracing::warn!(message = "INSIDE THE CAVE B");
        });
    });

    tracing::warn!(message = "OUTSIDE");

    // Don't exit
    loop {}
}
```

This seems to work without cloning of the base object? If you can confirm I'll get the version bumped to 0.3.0 and push it. Thanks for the patience.

Reply:
I'll get a test done this week. Thanks!